source/common/network/utility.cc

```diff
@@ -37,8 +38,8 @@ Address::InstanceConstSharedPtr Utility::resolveUrl(const std::string& url) {
   } else if (urlIsUdpScheme(url)) {
     return parseInternetAddressAndPort(url.substr(UDP_SCHEME.size()));
   } else if (urlIsUnixScheme(url)) {
-    return Address::InstanceConstSharedPtr{
-        new Address::PipeInstance(url.substr(UNIX_SCHEME.size()))};
+    return Address::InstanceFactory::InstanceFactory::createInstancePtrOrThrow<
```

Comment:
Put ASSERT(Thread::MainThread::isMainThread()) at the top of this function.

Reply:
Thanks, will do.